### PR TITLE
Text generation

### PR DESCRIPTION
**Pull request by connorcoley**
Addresses issue #101

This is still a WIP, but what do you think about the approach in general? Jinja is a nice way to template text (highly recommend it for the HTML table-type generation as well).

Tests, etc. are obvious missing, but here is an example output using a binary dump of the first Nielsen reaction example.

```
$ python ord_schema/generate_text.py 
    --input_file="examples/2_Nielsen_Deoxyfluorination_Screen/ord-nielsen-example.pb"
```

To a 1 mL glass vial was added (1) 0.1 mmol "c1ccccc1CCC(O)C" as a limiting reactant + 125 uL
 THF "C1CCCO1" as a solvent (dried); (2) 0.11 mmol "Clc1ccc(S(=O)(=O)F)cc1" as a reactant + 125
 uL THF "C1CCCO1" as a solvent (dried); and (3) 0.15 mmol MTBD "CN1CCCN2C1=NCCC2" as a
 reagent (purchased from Sigma-Millipore). The reaction was run under ambient temperature 
conditions. The reaction was run under air after fully sealing the reaction vessel. The reaction 
mixture was stirred at a high rate (600 rpm) using a stir bar. The reaction was analyzed after 48 h
 by NMR (19F NMR using 1 equiv 1-fluoronaphthalene in 250 uL chloroform as internal standard).
 "c1ccccc1CCC(F)C" was identified through NMR with a yield of 54% (p/m 4.8%) (assayed 
through NMR).

Another example using the Liu et al. OrgSyn reaction:

```
$ python ord_schema/generate_text.py --input_file="examples/3_Liu_Copper_OrgSyn/ord-liu-example.pb" 
```

To an oven-dried (oven dried and flushed with nitrogen) 500 mL glass round bottom flask (single-necked) was added (1) 21.3 g N,N-dibenzylhydroxylamine "C1=CC=C(C=C1)CN(CC2=CC=CC=C2)O" as a limiting reactant (purchased from TCI America, a few colored or darker crystals, which were present in trace amounts, were discarded using standard tweezers); (2) 12.8 g 4-dimethyl-aminopyridine "n1ccc(N(C)C)cc1" as a reactant (purchased from Sigma Aldrich, used as received, used as received); (3) 250 mL dichloromethane "C(Cl)Cl" as a solvent (purchased from J. T. Baker, dried, purified by passage under argon pressure through two packed columns of neutral alumina and copper(II) oxide); and (4) 12.9 mL pivaloyl chloride "CC(C)(C)C(=O)Cl" as a reactant (purchased from Alfa Aesar, used as received, used as received) (dropwise, over 5 min). The reaction was run in an ice bath (reaction started in ice bath before addition of pivaloyl chloride, then allowed to warm to 23 C, then stirred for additional 6 hours at RT (23 degC)) with a setpoint of 0 C. The reaction was run under nitrogen (dry nitrogen) using a needle to pierce the vessel septum (sealed with rubber septum, pierced with needle connected to manifold). The reaction mixture was stirred using a stir bar. The workup procedure consisted of (1) <WORKUP_STEP>; (2) <WORKUP_STEP>; (3) <WORKUP_STEP>; (4) <WORKUP_STEP>; (5) <WORKUP_STEP>; (6) <WORKUP_STEP>; (7) <WORKUP_STEP>; (8) <WORKUP_STEP>; (9) <WORKUP_STEP>; (10) <WORKUP_STEP>; and (11) <WORKUP_STEP> The reaction was analyzed after 6 h by NMR (101 MHz, CDCl3), NMR (400 MHz, CDCl3), IR spectroscopy (neat film, NaCl), high-resolution mass spectrometry (ESI-TOF), NMR (Quantitative NMR using 1,1,2,2-tetrachloroethane (>98%, purchased from Alfa Aesar) in CDCl3), and isolated weight (27.5-28.0 g recovered after workup). N,N-Dibenzyl-O-pivaloylhydroxylamine "O=C(C(C)(C)C)ON(CC1=CC=CC=C1)CC2=CC=CC=C2" was identified through NMR, NMR, and high-resolution mass spectrometry with a yield of 93.5% (p/m 0.5%) (assayed through isolated weight) with a purity of 99% (assayed through NMR). It appeared as a white powder. Note: the conditions for this reaction may not precisely fit the schema, so additional care should be taken when reading this description. Condition details: reaction started in ice bath before addition ofpivaloyl chloride, then allowed to warm to 23 C, then stirred for 'an additional 6 hours at RT (23 degC).

### CODE REVIEW

**Comment by skearnes** [comment]
This is awesome! Very nice.

**Comment by skearnes** [comment]
Perhaps include a provenance section, like "this description was automatically generated from ord-XXX, available at URL"

**Comment by connorcoley** [comment]
I added a very simple test for text generation. Since it is likely to evolve over the next few weeks/months, I don't want to write particularly stringent tests yet. 

>Perhaps include a provenance section, like "this description was automatically generated from ord-XXX, available at URL"

Also deferring this suggestion until we come up with the permalink strategy

File: ord_schema/visualization/drawing.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
+"""This module contains two molecular drawing functions to render SVGs or PNGs
```
Comment (skearnes) [inline ord_schema/visualization/drawing.py:1]: Should these files be in a separate PR?

Comment (connorcoley) [reply ord_schema/visualization/drawing.py:1]: The text generation and HTML generation uses the same base code. The drawing module is needed for HTML generation. Should I separate them?

Comment (skearnes) [reply ord_schema/visualization/drawing.py:1]: I think that might be good style, but I'm not going to require it to merge. You could have the text generation module output either txt or html and then have the drawing module use the html output for its own stuff; we might want to use the html in other settings as well...
